packages/sampler/src/MoonstoneEnvironment/MoonstoneEnvironment.js

```diff
@@ -86,44 +89,53 @@ const backgroundLabels = {
-	'backgroundColorful5': {background: '#547460 url("https://picsum.photos/1280/720?image=1053") no-repeat center/cover'},
-	'backgroundColorful6': {background: '#7c4590 url("https://picsum.photos/1280/720?image=967") no-repeat center/cover'},
-	'backgroundColorful7': {background: '#5d6542 url("https://picsum.photos/1280/720?image=1025") no-repeat center/cover'},
-	'backgroundColorful8': {background: '#555 url("https://picsum.photos/1280/720") no-repeat center/cover'}
+	'': '',
+	'backgroundColorful1': '#bb3352 url("http://picsum.photos/1280/720?image=1080") no-repeat center/cover',
+	'backgroundColorful2': '#4e6a40 url("http://picsum.photos/1280/720?image=1063") no-repeat center/cover',
+	'backgroundColorful3': '#5985a8 url("http://picsum.photos/1280/720?image=930") no-repeat center/cover',
+	'backgroundColorful4': '#71736d url("http://picsum.photos/1280/720?image=1044") no-repeat center/cover',
+	'backgroundColorful5': '#547460 url("http://picsum.photos/1280/720?image=1053") no-repeat center/cover',
+	'backgroundColorful6': '#7c4590 url("http://picsum.photos/1280/720?image=967") no-repeat center/cover',
+	'backgroundColorful7': '#5d6542 url("http://picsum.photos/1280/720?image=1025") no-repeat center/cover',
+	'backgroundColorful8': '#555 url("http://picsum.photos/1280/720") no-repeat center/cover'
 };
 
 const skins = {
 	'Dark': 'dark',
 	'Light': 'light'
 };
 
-// NOTE: Knobs cannot set locale in fullscreen mode. This allows any knob to be taken from the URL.
-const getPropFromURL = (propName, fallbackValue) => {
-	propName = encodeURI(propName);
-	const locationParams = window.parent.location.search;
+const getArgs = (str) => {
+	return qs.parse(str || (typeof window !== 'undefined' ? window.parent.location.search : ''));
+};
 
-	const startIndex = locationParams.indexOf('knob-' + propName + '=');
-	if (startIndex > -1) {
-		const keyIndex = locationParams.indexOf('=', startIndex);
+// This allows any prop/argument to be taken from the URL.
+const getPropFromArgs = (args, propName, fallbackValue) => {
+	if (args && typeof args[propName] !== 'undefined') {
+		return args[propName];
+	}
+	return fallbackValue;
+};
 
-		if (locationParams.indexOf('&', keyIndex) > -1 ) {
-			const valueIndex = locationParams.indexOf('&', keyIndex);
-			return locationParams.substring(keyIndex + 1, valueIndex);
-		} else {
-			return locationParams.substring(keyIndex + 1, locationParams.length);
+// This allows any knob to be taken from the URL.
+const getKnobFromArgs = (args, propName, fallbackValue) => {
+	const knobVal = getPropFromArgs(args, 'knob-' + propName, fallbackValue);
+	try {
+		// If it's valid JSON, parse it and return
+		return JSON.parse(knobVal);
+	} catch (e) {
+		// If it's invalid, see what it is, and in the unknown case, just return it (string most likely).
+		if (typeof knobVal === 'undefined' || knobVal === 'undefined' || knobVal === 'void 0') {
+			return void 0;
```

Comment:
Is it important that it be `undefined` vs `null`? If so, would it be more readable to `return undefined;` with an eslint pragma to suppress the error?

Reply:
Well, the two are different. To me, undefined means it was never set, and null means it was intentionally set to be blank or no value. I wanted it to be possible to explicitly set `null` and have that persist through this generalized method.
I didn't use `undefined` because I don't like adding linting exceptions. However, in this case, it may make the output easier to read/understand.

Reply:
I agree with your point on the difference between `undefined` and `null` and that using `return undefined` would be more readable. Let's make that update.

Reply:
You might be able to avoid this entire debate by refactoring the code a bit. I noticed that `getPropFromArgs` can return the fallback value but that value is then passed through `JSON.parse`. If the default value were a parsable string (not sure why it would be but nonetheless), the incorrect fallback would be used (the parsed value rather than the actual string value).

Consider the following which starts with the fallback value and only updates to the parsed value if the knob exists in `args` and can be parsed. The additional advantage being that if `fallbackValue` is `undefined`, we'll correctly return `undefined` without the eslint gymnastics.

```

// This allows any knob to be taken from the URL.
const getKnobFromArgs = (args, propName, fallbackValue) => {
	const knob = 'knob-' + propName;
	let value = fallbackValue;

	if (args && knob in args) {
		try {
			// If it's valid JSON, parse it
			value = JSON.parse(args[knob]);
		} catch (e) {
			// no handling required; allow fallbackValue to be used
		}
	}

	return value;
};
```

Reply:
This is simpler. I'll add it.